primedev/mods/modmanager.cpp

```diff
@@ -493,66 +386,27 @@ void ModManager::LoadMods()
 				}
 			}
 		}
-	}
 
-	// If there are new mods, we write entries accordingly in enabledmods.json
-	if (newModsDetected)
-	{
-		std::ofstream writeStream(cfgPath);
-		rapidjson::OStreamWrapper writeStreamWrapper(writeStream);
-		rapidjson::PrettyWriter<rapidjson::OStreamWrapper> writer(writeStreamWrapper);
-		m_EnabledModsCfg.Accept(writer);
-	}
-
-	// in a seperate loop because we register mod files in reverse order, since mods loaded later should have their files prioritised
-	for (int64_t i = m_LoadedMods.size() - 1; i > -1; i--)
-	{
-		if (!m_LoadedMods[i].m_bEnabled)
-			continue;
-
-		if (fs::exists(m_LoadedMods[i].m_ModDirectory / MOD_OVERRIDE_DIR))
+		// register mod files
+		if (fs::exists(mod.m_ModDirectory / MOD_OVERRIDE_DIR))
 		{
-			for (fs::directory_entry file : fs::recursive_directory_iterator(m_LoadedMods[i].m_ModDirectory / MOD_OVERRIDE_DIR))
+			for (fs::directory_entry file : fs::recursive_directory_iterator(mod.m_ModDirectory / MOD_OVERRIDE_DIR))
 			{
 				std::string path =
-					g_pModManager->NormaliseModFilePath(file.path().lexically_relative(m_LoadedMods[i].m_ModDirectory / MOD_OVERRIDE_DIR));
+					g_pModManager->NormaliseModFilePath(file.path().lexically_relative(mod.m_ModDirectory / MOD_OVERRIDE_DIR));
 				if (file.is_regular_file() && m_ModFiles.find(path) == m_ModFiles.end())
 				{
 					ModOverrideFile modFile;
-					modFile.m_pOwningMod = &m_LoadedMods[i];
+					modFile.m_pOwningMod = &mod;
 					modFile.m_Path = path;
-					m_ModFiles.insert(std::make_pair(path, modFile));
+					m_ModFiles.insert_or_assign(path, modFile);
```

Comment:
I changed the logic here a bit, before we were using `insert` and reversing the loop, this meant that mods at the end of the list were preferred. `insert_or_assign` in a non-reversed list should achieve the same result and avoid us looping over the list again.